packages/frontend/src/features/MetricFlow/hooks/useMetricFlowFieldsAPI.ts

```diff
@@ -0,0 +1,41 @@
+import { ApiError } from '@lightdash/common';
+import { uniqWith } from 'lodash-es';
+import { useQuery } from 'react-query';
+import { UseQueryOptions } from 'react-query/types/react/types';
+import {
+    getMetricFlowFields,
+    GetMetricFlowFieldsResponse,
+} from '../../../api/MetricFlowAPI';
+
+export const useMetricFlowFieldsAPI = (
+    projectUuid?: string,
+    filters?: {
+        metrics: string[];
+        dimensions: string[];
+    },
+    useQueryOptions?: UseQueryOptions<GetMetricFlowFieldsResponse, ApiError>,
+) => {
+    return useQuery<GetMetricFlowFieldsResponse, ApiError>({
+        queryKey: ['metric_flow_fields', projectUuid, filters],
+        enabled: !!projectUuid,
+        queryFn: () => getMetricFlowFields(projectUuid!, filters),
+        keepPreviousData: true,
+        select: (data) => {
+            // If no dimensions are returned, use the dimensions from the metrics
+            if (data.dimensions.length === 0) {
```

Comment:
The current logic acts as a fallback since it only depends on the data returned. If the logic is based on the inputs (filters) it will be business logic.